packages/postgres-database/src/simcore_postgres_database/utils_folders.py

```diff
@@ -511,47 +570,71 @@ async def folder_create(
         )
         if entry_exists:
             raise FolderAlreadyExistsError(
-                product_name=product_name, folder=name, parent=parent, gid=gid
+                product_name=product_name, folder=name, parent=parent, gids=gids
             )
 
+        # `permissions_gid` is compted as follows:
+        # - `folder has a parent?` taken from the resolved access rights of the parent folder
+        # - `is root folder, a.k.a. no parent?` taken from the user's primary group
+        permissions_gid: _GroupID | None = None
         if parent:
-            # check if parent has permissions
-            await _check_folder_and_access(
+            resolved_access_rights = await _check_folder_and_access(
                 connection,
                 product_name,
                 folder_id=parent,
-                gid=gid,
+                gids=gids,
                 permissions=_required_permissions,
                 enforece_all_permissions=False,
             )
+            permissions_gid = resolved_access_rights.gid
+
+        if permissions_gid is None:
+            groups_results: list[RowProxy] | None = await (
+                await connection.execute(
+                    sa.select([groups.c.gid, groups.c.type]).where(
+                        groups.c.gid.in_(gids)
+                    )
+                )
+            ).fetchall()
+
+            if not groups_results:
+                raise GroupIDsDoNotExistError(gids=gids)
+
```

Comment:
again. is there nothing groups module that you can reuse here? you are listing the groups of the user??